### PR TITLE
pre-commit hook + minor fixes

### PR DESCRIPTION
**Pull request by romainx**
# pre-commit

[pre-commit](https://pre-commit.com/) added to check

- `bash` files with bashate see #1109
- `yaml` files with yamlint see #1108
- `docker` files with hadolint see #1100

# hadolint

Version of the linter added as a variable in the `Makefile`

Could be adapted to integrate python file check see #1118

### CODE REVIEW

**Comment by romainx** [comment]
Build error has nothing to do with the change. It has occurred in `r-notebook` during the import of `tidymodels`

```
AssertionError: {'tidymodels': AssertionError('Package [tidymodels] import failed\nassert 1 == 0\n  -1\n  +0')}
```

I will have a look.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.1.0
+    hooks:
+      #- id: flake8
```
Comment (mathbunnyru) [inline .pre-commit-config.yaml:6]: Seems to be some debug info

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - id: check-yaml
+        files: .*\.(yaml|yml)$
+  - repo: https://github.com/adrienverge/yamllint.git
+    rev: v1.23.0
+    hooks:
+      - id: yamllint
+        files: \.(yaml|yml)$
+        exclude: ^.travis
```
Comment (mathbunnyru) [inline .pre-commit-config.yaml:14]: I think travis works fine with formatted yaml files, so I don't think we need an exclude

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  - repo: https://github.com/openstack-dev/bashate.git
+    rev: 2.0.0
+    hooks:
+      - id: bashate
+  - repo: local
+    hooks:
+      - id: hadolint
```
Comment (mathbunnyru) [inline .pre-commit-config.yaml:21]: I see some inconsistency between hook id "hadolint" and the command "lint-all".
Maybe we should rename something?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (romainx) [reply .pre-commit-config.yaml]: Removed it was to thing about adding flake8 see  #1118

File: .pre-commit-config.yaml (hunk 3 above)
Comment (romainx) [reply .pre-commit-config.yaml:20]: You're right, fixed

File: .pre-commit-config.yaml (hunk 2 above)
Comment (romainx) [reply .pre-commit-config.yaml]: The `.travis.yml` has some lines too long. I've adapted the configuration of `yamlint` to relax this constraint.

**Comment by romainx** [comment]
@mathbunnyru I've made some ajustements following your review. 
I would like to have your opinion on this new version.
Thanks in advance.

**Comment by mathbunnyru** [comment]
Thank you @romainx !
I took a look after your changes.

LGTM, but I can think about 2 things which may be good:
1. Run pre-commit hook server-side as well. So hadolint is checked server-side, but other linters are not.
2. Add some info for developers to make it easier for newcomers to install these tools and run them.

It can be done in a separate PR, up to you to do decide how you want to implement this.

**Comment by romainx** [comment]
@mathbunnyru thank you, once again good suggestions 🥇.

1. Good idea, however I prefer managing that in a different PR since it will involve CI build debugging. I would prefer only bootstrap pre-commit in this PR and improve its usage over time -- no it's not laziness 🤣.
2. I take it in this PR! I was just thinking about it, there is currently no dedicated page, but I intend to adjust page `lint.md` to talk about this.

For the point 2 I let you know when it's done. I will try to do it quickly.

Once again thanks for you constructive feedback 😀

**Comment by romainx** [comment]
@mathbunnyru I'm good with the doc. Please tell me if it's Ok for you.

**Comment by mathbunnyru** [comment]
Good job, @romainx!
Everything is great 👍

**Comment by parente** [comment]
Thanks for your patience on this PR. I've resolved the merge conflict with the master branch. I'll merge when CI succeeds.

**Comment by parente** [comment]
Thanks for the PR, @romainx and the review @mathbunnyru.

**Pull request merged**